docs/proposals/multi-node/README.md

```diff
@@ -866,15 +899,15 @@ Mark the `n - s` members (corresponding to the pods that are not `Ready`) as `No
     - Ready: `n - f`
     - Members `NotReady` for long enough to be evicted, i.e. `lastTransitionTime > notReadyGracePeriod`: `f` where `f < n/2`
     - Members with readiness status `Unknown` long enough to be considered `NotReady`, i.e. `lastTransitionTime > unknownGracePeriod`: `0`
-    - Members with heartbeat stale enough to be considered as of `Unknown` readiness status, i.e. `lastHeartbeatTime > heartbeatGracePeriod`: N/A
+    - Members with expired `lease`: N/A
   - conditions:
     - Ready: true
     - AllMembersReady: false
     - BackupReady: true
 
 #### Recommended Action
 
-Delete the `f` `NotReady` member pods to force restart of the pods if they do not automatically restart via failed `livenessProbe`. The expectation is that they will either re-join the cluster as an existing member or remove themselves and join as new members on restart of the container or pod.
+Delete the `f` `NotReady` member pods to force restart of the pods if they do not automatically restart via failed `livenessProbe`. The expectation is that they will either re-join the cluster as an existing member or remove themselves and join as new members on restart of the container or pod and [renew their `leases`](#member-leases).
```

Comment:
Should Druid reclaim the identity of the member `lease` when we delete the pod? A restart usually takes longer and otherwise during the next health check, we wouldn't know if the restart already finished.

Reply:
By _reclaiming the identity_ if you meant recreating the `lease` object (or maybe resetting `spec.holderIdentity`), I was thinking about this while writing the change but couldn't conclude either way 😅 

When the pod is restarted, member ID (and hence, `spec.holderIdentity`)  would change only if the data and/or WAL files are lost. Otherwise, the member would simply adopt the older identity (probably, most frequent case while using PVCs for the member pods). We could think of resetting `spec.holderIdentity` before restarting the pod and adding an additional check in `etcd-backup-restore` to renew the lease only if `spec.holderIdentity` is blank or the same as the ETCD member id. This might strengthen some edge-cases where the previous member pod is still somehow running. But I thought `StatefulSet` controller has this already under controller and so, was not sure if we need the additional check. WDYT?

Reply:
> By reclaiming the identity if you meant recreating the lease object (or maybe resetting spec.holderIdentity)

Yes, that's what I meant. But the reason for this was not that a previous pod could still run, rather the following:
1. Lease is expired & not renewed
2. Status turns `UNKNOWN`
3. Status turns `FALSE` after `etcd-member-notready-threshold`
4. Custodian deletes the pod
5. Lease from (1.) is still expired
6. Custodian checks lease again **before** new pod renews the lease
7. Custodian deletes the pod again

Reply:
> Custodian checks lease again before new pod renews the lease
> Custodian deletes the pod again

I thought this could be avoided by checking if `creationTimestamp` of the pod is after the `lastTransitionTime` for its member (or `renewTime` of its `lease`). Or do you see the `Get` call on pod a unnecessary?

Reply:
I'm fine with both ways, just wanted to clarify how we're going to tackle this issue 👍